components/org.wso2.carbon.identity.api.server.notification.sender/org.wso2.carbon.identity.api.server.notification.sender.v2/src/main/java/org/wso2/carbon/identity/api/server/notification/sender/v2/core/NotificationSenderManagementService.java

```diff
@@ -0,0 +1,493 @@
+
+        if (dto.getProperties() == null) {
+            return emailSender;
+        }
+        dto.getProperties().forEach((key, value) -> {
+            if (StringUtils.isBlank(key) || StringUtils.isBlank(value) || excludedKeys.contains(key)) {
+                return;
+            }
+            Properties prop = new Properties();
+            prop.setKey(key);
+            prop.setValue(value);
+            properties.add(prop);
+        });
+        emailSender.setProperties(properties);
+        return emailSender;
+    }
+
+    private SMSSenderDTO buildSMSSenderDTO(SMSSenderAdd smsSenderAdd) {
+
+        SMSSenderDTO dto = new SMSSenderDTO();
+        dto.setName(smsSenderAdd.getName());
+        dto.setProvider(smsSenderAdd.getProvider());
+        dto.setProviderURL(smsSenderAdd.getProviderURL());
+        dto.setKey(smsSenderAdd.getKey());
+        dto.setSecret(smsSenderAdd.getSecret());
+        dto.setSender(smsSenderAdd.getSender());
+        dto.setContentType(smsSenderAdd.getContentType().toString());
+        List<Properties> properties = smsSenderAdd.getProperties();
+        if (properties != null) {
+            properties.forEach((prop) -> dto.getProperties().put(prop.getKey(), prop.getValue()));
+        }
+        return dto;
+    }
+
+
+    private SMSSenderDTO buildSMSSenderDTO(String senderName, SMSSenderUpdateRequest smsSenderUpdateRequest) {
+
+        SMSSenderDTO dto = new SMSSenderDTO();
+        dto.setName(senderName);
+        dto.setProvider(smsSenderUpdateRequest.getProvider());
+        dto.setProviderURL(smsSenderUpdateRequest.getProviderURL());
+        dto.setKey(smsSenderUpdateRequest.getKey());
+        dto.setSecret(smsSenderUpdateRequest.getSecret());
+        dto.setSender(smsSenderUpdateRequest.getSender());
+        dto.setContentType(smsSenderUpdateRequest.getContentType().toString());
+        List<Properties> properties = smsSenderUpdateRequest.getProperties();
+        if (properties != null) {
+            properties.forEach((prop) -> dto.getProperties().put(prop.getKey(), prop.getValue()));
+        }
+        return dto;
+    }
+
+    private APIError handleException(NotificationSenderManagementException e) {
+
+        if (e instanceof NotificationSenderManagementClientException) {
+            throw buildClientError(e);
+        }
+        throw buildServerError(e);
+    }
+
+    private SMSSender buildSMSSenderFromDTO(SMSSenderDTO dto) {
+
+        SMSSender smsSender = new SMSSender();
+        smsSender.setName(dto.getName());
+        smsSender.setProvider(dto.getProvider());
+        smsSender.setProviderURL(dto.getProviderURL());
+        smsSender.setKey(dto.getKey());
+        smsSender.setSecret(dto.getSecret());
+        smsSender.setSender(dto.getSender());
+        smsSender.setContentType(SMSSender.ContentTypeEnum.valueOf(dto.getContentType()));
+        List<Properties> properties = new ArrayList<>();
+        dto.getProperties().forEach((key, value) -> {
```

Comment:
Null safe ?

Reply:
Not relevant since this PR focuses only on EmailSender improvements